src/backend/base/langflow/custom/custom_component/component.py

```diff
@@ -830,4 +837,34 @@ def log(self, message: LoggableType | list[LoggableType], name: str | None = Non
 
     def _append_tool_output(self):
         if next((output for output in self.outputs if output.name == TOOL_OUTPUT_NAME), None) is None:
-            self.outputs.append(Output(name=TOOL_OUTPUT_NAME, display_name="Tool", method="to_toolkit", types=["Tool"]))
+            set_as_tool_input = self._inputs.get("set_as_tool")
+            hidden_value = not set_as_tool_input.value if set_as_tool_input is not None else False
+            self.outputs.append(
+                Output(
+                    name=TOOL_OUTPUT_NAME,
+                    display_name="Tool",
+                    method="to_toolkit",
+                    types=["Tool"],
+                    hidden=hidden_value,
+                )
+            )
+
+    def update_build_config(self, build_config: dotdict, field_value: Any, field_name: str | None = None):
+        return super().update_build_config(build_config, field_value, field_name)
+
+    def _append_tool_toggle_output(self):
+        from langflow.inputs import BoolInput
+
+        self.inputs.append(
+            BoolInput(
+                name="set_as_tool",
+                value=False,
+                display_name="Component as Tool",
+                advanced=True,
+                info="If True, the component will be treated as a tool.",
+                onChange=lambda value: self.update_build_config(
+                    build_config=self.build_config(), field_value=value, field_name="set_as_tool"
+                ),
+            )
+        )
+        # TODO: update Frontend outputs accoriding to the toggle may be by updating the build config
```

Comment:
Is this necessary work for this PR, or will it be done in a follow up?

Reply:
for the further PR

Reply:
added as TODO